flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManagerTest.java

```diff
@@ -470,6 +483,57 @@ private void testNotificationAboutNotEnoughResources(boolean withNotificationGra
         };
     }
 
+    /**
+     * Test that checkResourceRequirements will only be triggered once after multiple trigger
+     * function calls.
+     */
+    @Test
+    public void testRequirementCheckOnlyTriggeredOnce() throws Exception {
+        new Context() {
+            {
+                final AtomicInteger checkTimes = new AtomicInteger();
+                final long requirementCheckDelay = 20;
+                resourceAllocationStrategyBuilder.setTryFulfillRequirementsFunction(
+                        (ignored1, ignored2) -> {
+                            checkTimes.addAndGet(1);
+                            return ResourceAllocationResult.builder().build();
+                        });
+                setRequirementCheckDelay(requirementCheckDelay);
+                runTest(
+                        () -> {
+                            final ResourceRequirements resourceRequirements1 =
+                                    createResourceRequirementsForSingleSlot();
+                            final ResourceRequirements resourceRequirements2 =
+                                    createResourceRequirementsForSingleSlot();
+                            final ResourceRequirements resourceRequirements3 =
+                                    createResourceRequirementsForSingleSlot();
+                            final TaskExecutorConnection taskExecutionConnection =
+                                    createTaskExecutorConnection();
+                            getSlotManager().processResourceRequirements(resourceRequirements1);
+                            getSlotManager().processResourceRequirements(resourceRequirements2);
+                            getSlotManager()
+                                    .registerTaskManager(
+                                            taskExecutionConnection,
+                                            new SlotReport(),
+                                            DEFAULT_TOTAL_RESOURCE_PROFILE,
+                                            DEFAULT_SLOT_RESOURCE_PROFILE);
+                            Thread.sleep(requirementCheckDelay * 2);
```

Comment:
I would suggest to use `ComletableFuture#get(timeout, unit)` instead of sleep. The future may complete earlier than the timeout.